src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -125,16 +126,46 @@ public ShooterSubsystem(final ShooterHoodSubsystem shooterHood, ShuffleboardTab
 
     NetworkTableEntry shooterPIDchooser = tab.add("Shooter PID", true).withWidget(BuiltInWidgets.kToggleSwitch)
         .getEntry();
+
     shooterPIDchooser.addListener(notice -> enablePID = notice.value.getBoolean(),
         EntryListenerFlags.kNew | EntryListenerFlags.kUpdate);
+
+    // Configures a Shuffleboard widget for flywheel speed. It will display as a
+    // graph of the flywheel speed from the last 20 seconds.
+    flywheelSpeed = tab.add("Flywheel Speed (RPM)", 0).withWidget(BuiltInWidgets.kNumberBar).getEntry();
+
+    // Adds a Shuffleboard widget to show whether the flywheel is spinning within a
+    // certain tolerance of the setpoint. See isFlywheelReady().
+    isFlywheelReady = tab.add("Is Flywheel Ready?", false).withWidget(BuiltInWidgets.kBooleanBox).getEntry();
+
+    shooterSetpoint = tab.addPersistent("Shooter Setpoint", 4500).withWidget(BuiltInWidgets.kTextView).getEntry();
+    shooterSetpoint.addListener(notice -> setpoint = notice.value.getDouble(),
+        EntryListenerFlags.kNew | EntryListenerFlags.kUpdate);
+
+    kPSlider = tab.addPersistent("Shooter kP", 5e-5).withWidget(BuiltInWidgets.kTextView).getEntry();
+    kPSlider.addListener(notice -> kP = notice.value.getDouble(), EntryListenerFlags.kNew | EntryListenerFlags.kUpdate);
+
+    kISlider = tab.addPersistent("Shooter kI", 1e-6).withWidget(BuiltInWidgets.kTextView).getEntry();
+    kISlider.addListener(notice -> kI = notice.value.getDouble(), EntryListenerFlags.kNew | EntryListenerFlags.kUpdate);
+
+    kIZSlider = tab.addPersistent("Shooter kIZ", 1e-6).withWidget(BuiltInWidgets.kTextView).getEntry();
+    kIZSlider.addListener(notice -> kIZ = notice.value.getDouble(),
+        EntryListenerFlags.kNew | EntryListenerFlags.kUpdate);
+
+    kDSlider = tab.addPersistent("Shooter kD", 0).withWidget(BuiltInWidgets.kTextView).getEntry();
+    kDSlider.addListener(notice -> kD = notice.value.getDouble(), EntryListenerFlags.kNew | EntryListenerFlags.kUpdate);
+
+    kFFSlider = tab.addPersistent("Shooter kFF", 0.000156).withWidget(BuiltInWidgets.kTextView).getEntry();
+    kFFSlider.addListener(notice -> kFF = notice.value.getDouble(),
+        EntryListenerFlags.kNew | EntryListenerFlags.kUpdate);
   }
 
   public void shoot(double hoodSetpoint) {
     // fast shooter wheel
     // deploy hood
     // run gate if both of "" are good
 
-    double pidSetpoint = MoPrefs.getShooterPIDSetpoint();
```

Comment:
What the heck is going on here?  The setpoint should be a MoPref and not hardcoded (aside from a default).